homeassistant/components/smarttub/config_flow.py

```diff
@@ -10,44 +10,84 @@
+                )
+
+            except LoginFailed:
+                errors["base"] = "invalid_auth"
+            else:
+                await self.async_set_unique_id(account.id)
+
+                if self._reauth_input is None:
+                    self._abort_if_unique_id_configured()
+                    return self.async_create_entry(
+                        title=user_input[CONF_EMAIL], data=user_input
+                    )
+
+                # this is a reauth attempt
+                if self._reauth_entry.unique_id != self.unique_id:
+                    # there is a config entry matching this account, but it is not the one we were trying to reauth
+                    return self.async_abort(reason="already_configured")
+                self.hass.config_entries.async_update_entry(
+                    self._reauth_entry, data=user_input
+                )
+                await self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
 
-        controller = SmartTubController(self.hass)
-        try:
-            account = await controller.login(
-                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+    async def async_step_reauth(self, user_input=None):
+        """Get new credentials if the current ones don't work anymore."""
+        self._reauth_input = dict(user_input)
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            # same as DATA_SCHEMA but with default email
+            data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        CONF_EMAIL, default=self._reauth_input.get(CONF_EMAIL)
+                    ): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
             )
-        except LoginFailed:
-            errors["base"] = "invalid_auth"
             return self.async_show_form(
-                step_id="user", data_schema=DATA_SCHEMA, errors=errors
+                step_id="reauth_confirm",
+                data_schema=data_schema,
             )
-
-        existing_entry = await self.async_set_unique_id(account.id)
-        if existing_entry:
-            self.hass.config_entries.async_update_entry(existing_entry, data=user_input)
-            await self.hass.config_entries.async_reload(existing_entry.entry_id)
-            return self.async_abort(reason="reauth_successful")
-
-        return self.async_create_entry(title=user_input[CONF_EMAIL], data=user_input)
+        return await self.async_step_user()
```

Comment:
```suggestion
        return await self.async_step_user(user_input)
```

If you don't pass the input, its going to ask again.

`python3 -m script.translations develop` to regen the `en.json`

Reply:
Thanks. This bit is also verbatim from https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication, I'm still a bit unclear on how all of this is wired together. If I figure it out I'll submit a PR for the docs